purchase_product_barcode/reports/purchase_order_templates.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_purchaseorder_document"
+              inherit_id="purchase.report_purchaseorder_document">
+        <xpath expr="//table[@class='table table-condensed']/thead/tr/th[1]"
+                   position="after">
+            <th><strong>Barcode</strong></th>
+        </xpath>
+        <xpath expr="//table[@class='table table-condensed']/tbody/tr/td[1]"
+                   position="after">
+            <td>
+                <span t-if="line.barcode">
+                    <img t-att-src="'/report/barcode/?type=%s&amp;value=%s&amp;width=%s&amp;height=%s'
+                                    % ('EAN13', line.barcode, 600, 100)"
```

Comment:
Odoo does not support 'JAN' barcode directly but 'JAN' is compatible with 'EAN13'.